### DISCUSSION TITLE
URL Recursive Scraping & Conversion

### DISCUSSION THREAD

**Discussion by jethrop** [cline/cline#494] [Feature Requests]
# Context

It is my understanding that when the @ url function is used it only scrapes the supplied URL and doesn't recursively move down the directory structure or follow links.  This works for some simple instances but doesn't work when the needed information is more nested, such as with documentation sites.  I understand the @ url feature was designed to partially solve the documentation problem (see issue #306).  As it stands, however, a lot of important information is getting missed.

# Request

Add an option to automatically scrape for deeper nested information.  This could be like an @ docs function that's specifically designed for this.
Alternatively, integrate a feature like [Firecrawl](https://github.com/mendableai/firecrawl) to automatically download and format the information for easier reference by the llm.  HTML -> MD -> LLM

_Apologies if this is a noob request.  Also apologies if I missed a similar request.  I searched through the requests related to "url" but didn't find one._

**Comment by jethrop** [comment]
This medium article explains why markdown is a good format for feeding LLMs: https://towardsdatascience.com/making-text-data-ai-ready-81d7fa83fd58

**Comment by jethrop** [comment]
After building and testing my own [scraper to markdown script](https://github.com/jethrop/scrapeToMarkdown) I believe it's much better to institute an indexing system like [Contine dev uses](https://docs.continue.dev/customize/deep-dives/docs).  The llm cannot handle the entire docs library at a time.

**Reply by jethrop** [reply]
Which is obvious in hindsight, yes.

**Reply by jethrop** [reply]
I believe this is the docs indexing code from the continue dev repo:
https://github.com/continuedev/continue/tree/main/core/indexing/docs

**Comment by TomLucidor** [comment]
Seconding this but there are enough people scared of the cybersec that this might take a little bit of a backseat, also always be ready to use an alt-library https://github.com/unclecode/crawl4ai
